packages/material-ui/src/transitions/utils.ts

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+
+export const reflow = (node: Element) => node.scrollTop;
+
+interface ComponentProps {
+  easing: string | { enter?: string; exit?: string };
+  style: React.CSSProperties;
```

Comment:
`style` seems to be optional
```suggestion
  style?: React.CSSProperties;
```

Reply:
Good point. Though we do want to require this prop but make it nullable.

I've come to treat `?` and `| undefined` differently (and [TypeScript does so as well starting with 4.4](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4-beta/#exact-optional-property-types))

The thing is that `getTransitionProps({ timeout, easing })` would be wrong. If your component has a custom inline style, `getTransitionProps` would override it. However, `getTransitionProps({ style, timeout, easing })` is correct. Even though the type of `style` is indistinguishable at runtime.